pageserver/src/tenant/timeline.rs

```diff
@@ -4572,6 +4573,22 @@ impl Timeline {
                     start = img_range.end;
                     continue;
                 }
+            } else if let ImageLayerCreationMode::Force = mode {
+                // When forced to create image layers, we might try and create them where they already
+                // exist.  This mode is only used in tests/debug.
+                let layers = self.layers.read().await;
+                if layers.contains_key(&PersistentLayerKey {
+                    key_range: img_range.clone(),
+                    lsn_range: PersistentLayerDesc::image_layer_lsn_range(lsn),
+                    is_delta: false,
+                }) {
```

Comment:
We are doing this query against the hashmap; this only works assuming the partitioning doesn't change. Should the query be done against layermap and image coverage?

Reply:
This is `#[cfg(feature = testing)]` afaik, so maybe something to consider. I do wonder if this means a bug with the overall force handling.